interfaces/builtin/hugepages_control.go

```diff
@@ -0,0 +1,73 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt
+// https://www.kernel.org/doc/Documentation/vm/transhuge.txt
+// This interface assumes that huge pages are mounted at either:
+// - /dev/hugepages (Debian, Ubuntu)
+// - /run/hugepages (various documentation)
+const hugepagesControlSummary = `allows controlling hugepages`
+
+const hugepagesControlBaseDeclarationSlots = `
+  hugepages-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const hugepagesControlConnectedPlugAppArmor = `
+# Allow configuring huge pages via /sys or /proc
+/sys/kernel/mm/hugepages/{,hugepages-[0-9]*}/* r,
+/sys/kernel/mm/hugepages/{,hugepages-[0-9]*}/nr_{hugepages,hugepages_mempolicy,overcommit_hugepages} w,
+/sys/devices/system/node/node[0-9]*/hugepages/{,hugepages-[0-9]*}/* r,
+/sys/devices/system/node/node[0-9]*/hugepages/{,hugepages-[0-9]*}/nr_hugepages w,
+@{PROC}/sys/vm/nr_{hugepages,hugepages_mempolicy,overcommit_hugepages} rw,
```

Comment:
There is another file in proc that is related to hugepages. If we don't want to expose control over that file perhaps it should be added as a commented-out entry with a suitable description? Also perhaps consider making that file readable for now.

```
# hugetlb_shm_group controls which groups can create shared memory segments
# TODO: consider enabling this when snapd groups are implemented.
# @{PROC}/sys/vm/hugetlb_shm_group r, 
```

Reply:
Making it readable can make sense, I'll let @jdstrand comment on that though.

Reply:
I don't think that applications will be checking this file, but read access is fine for this interface either now or when someone finds it is needed down the line. I like the suggested comment, but this feels like a code comment rather than a policy comment.

Reply:
I have added the read rule to the policy.